data_processing/cleaning_demographic_data.Rmd

```diff
@@ -52,11 +50,412 @@ demo_dat %>%
 Should we drop everything from " -" (i.e. space and then dash)
 ```{r}
 demo_dat %>%  
-  mutate(aps_report_location = str_replace(aps_report_location, "'", "")) %>% 
-  mutate(aps_report_location = str_replace(aps_report_location, ", ca|\\ ca|,ca", "")) %>%        count(aps_report_location) %>% filter(str_detect(aps_report_location, "-")) %>% 
+  count(aps_report_location) %>% 
+  filter(str_detect(aps_report_location, "-")) %>% 
   filter(str_detect(aps_report_location, "-"))
 ```
 
+replace all locations that contain Roseville in the name to just Roseville as a new location variable 
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(aps_location_recode = case_when(str_detect(aps_report_location,"rosev") ~ "roseville",
+                                      
+                                         TRUE ~ aps_report_location))
+```
+
+check
+```{r}
+demo_dat %>% 
+  count(aps_report_location, aps_location_recode) %>% 
+  filter(str_starts(aps_report_location, "r"))
+```
+Location of Participation:
+remove "'",", ca" and "ca" from the end of responses
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(location_of_participation_recode = str_replace(location_of_participation, "'", "")) %>% 
+  mutate(location_of_participation_recode = str_replace(location_of_participation, ", ca| ca|,ca", ""))
+```
+
+Check for removal of Ca
+```{r}
+demo_dat %>% 
+  filter(str_detect(location_of_participation, ",")) %>% 
+  count(location_of_participation, location_of_participation_recode)
+```
+
+remove special characters like ":" before laguna beach. Combine misspellings of bakersfield.Combine mispelling of barstow.
+```{r}
+demo_dat <- demo_dat %>% 
+  mutate(location_of_participation_recode = case_when(str_detect(location_of_participation, ":") ~ "laguna beach",
+                                                      str_detect(location_of_participation, "bakersf|bakerf") ~ "bakersfield",
+                                                      str_detect(location_of_participation, "bartso") ~ "barstow",
+                                                      TRUE ~ location_of_participation))
+```
+
```

Comment:
This code looks good. Add a sanity check right below using the `count()` function where you print the distribution of `location_of_participation` by the `location_of_participation_recode` For instance,

```
data %>%
count(og_var, recode_var)

```